server/mailet/mailets/src/test/java/org/apache/james/transport/mailets/remote/delivery/DeliveryRunnableTest.java

```diff
@@ -240,6 +245,29 @@ void deliveryTemporaryFailureShouldResetDeliveryCountOnNonErrorState() throws Ex
         verify(mailQueue).enQueue(FakeMail.builder()
                 .name("name")
                 .attribute(DeliveryRetriesHelper.makeAttribute(1))
+                .attribute(MailPrioritySupport.LOW_PRIORITY_ATTRIBUTE)
+                .attribute(new Attribute(IS_DELIVERY_PERMANENT_ERROR, AttributeValue.of(false)))
+                .state(Mail.ERROR)
+                .lastUpdated(FIXED_DATE)
+                .build(),
+            Duration.ofSeconds(1));
+        verifyNoMoreInteractions(mailQueue);
+    }
+
+    @Test
+    void deliveryTemporaryFailureShouldRetryDeliveryWithPreservedPriority() throws Exception {
```

Comment:
We should ensure that all outgoing components are able to set priorities based on the outgoing queue content too.

See https://github.com/apache/james-project/tree/master/server/mailet/remote-delivery-integration-testing for writing such tests.

I would expect that James transmit the mails precisely with the priority I did submit it with.